instrumentation-annotations-support/src/test/java/io/opentelemetry/instrumentation/api/annotation/support/MethodSpanAttributesExtractorTest.java

```diff
@@ -0,0 +1,181 @@
+
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.internal.cache.Cache;
+import java.lang.reflect.Method;
+import java.util.function.Function;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class MethodSpanAttributesExtractorTest {
+
+  Object request = new Object();
+  Context context = Context.root();
+  Method method = TestClass.getMethod();
+
+  @Mock AttributesBuilder builder;
+  @Mock Cache<Method, AttributeBindings> cache;
+
+  @BeforeEach
+  void setup() {
+    lenient()
+        .doAnswer(
+            invocation -> {
+              Method m = invocation.getArgument(0);
+              Function<Method, AttributeBindings> fn = invocation.getArgument(1);
+              return fn.apply(m);
+            })
+        .when(cache)
+        .computeIfAbsent(any(), any());
+  }
+
+  @Test
+  void extractAttributesForMethodWithAttributeNames() {
+    MethodSpanAttributesExtractor<Object, Object> extractor =
+        new MethodSpanAttributesExtractor<>(
+            r -> method,
+            (m, p) -> new String[] {"x", "y", "z"},
+            r -> new String[] {"a", "b", "c"},
+            cache);
+
+    extractor.onStart(builder, context, request);
+
+    verify(builder).put(stringKey("x"), "a");
+    verify(builder).put(stringKey("y"), "b");
+    verify(builder).put(stringKey("z"), "c");
+  }
+
+  @Test
+  void doesNotExtractAttributesForEmptyAttributeNameArray() {
+    MethodSpanAttributesExtractor<Object, Object> extractor =
+        new MethodSpanAttributesExtractor<>(
+            r -> method, (m, p) -> new String[0], r -> new String[] {"a", "b", "c"}, cache);
+
+    extractor.onStart(builder, context, request);
+    verifyNoInteractions(builder);
+  }
+
+  @Test
+  void doesNotExtractAttributesForMethodWithAttributeNamesArrayWithFewerElementsThanParams() {
+    MethodSpanAttributesExtractor<Object, Object> extractor =
+        new MethodSpanAttributesExtractor<>(
+            r -> method,
+            (m, p) -> new String[] {"x", "y"},
+            r -> new String[] {"a", "b", "c"},
+            cache);
+
+    extractor.onStart(builder, context, request);
+    verifyNoInteractions(builder);
+  }
+
+  @Test
+  void extractsAttributesForMethodWithAttributeNamesArrayWithNullElement() {
+    MethodSpanAttributesExtractor<Object, Object> extractor =
+        new MethodSpanAttributesExtractor<>(
+            r -> method,
+            (m, p) -> new String[] {"x", null, "z"},
+            r -> new String[] {"a", "b", "c"},
+            cache);
+
+    extractor.onStart(builder, context, request);
+
+    verify(builder).put(stringKey("x"), "a");
+    verify(builder).put(stringKey("z"), "c");
+    verify(builder, never()).put(anyString(), eq("b"));
```

Comment:
Shouldn't it be `any(AttributeKey.class)` instead of `anyString()`? Or maybe `verifyNoMoreInteractions` instead

Reply:
Yeah, there are so many overloads for `put()`, heh, that was wrong. Looks like `verifyNoMoreInteractions()` works here. 👍🏻